src/data_science/data_science.py

```diff
@@ -490,19 +432,127 @@ def get_verkehrsleistung_time_by_config_id(
+            data = [
+                entry.value
+                for entry in smard_api.get_data(
+                    strategy.start_datetime,
+                    strategy.start_datetime
+                    + timedelta(seconds=strategy.end_tick - strategy.start_tick),
+                )
+            ]
+            data = map(strategy.compute_coal_consumption, data)
+            smard_df = pd.DataFrame(data, columns=[f"value_{config_id}"])
+
+            smard_df["tick"] = pd.Series(
+                range(strategy.start_tick, strategy.end_tick + 1, 900), dtype="int64"
+            )
+            smard_df.set_index("tick", inplace=True)
+
+            dataframes.append(smard_df)
 
-    def get_spawn_events_by_config_id(self, config_id: UUID) -> pd.DataFrame:
+        result_df = pd.concat(dataframes)
+        result_df.reset_index(inplace=True)
+        result_df["time"] = result_df["tick"] + self.unix_2020
+        result_df["time"] = pd.to_datetime(result_df["time"], unit="s")
+        result_df.set_index("time", inplace=True)
+        del result_df["tick"]
+        return result_df
+
+    def get_coal_spawn_events_by_config_id(
+        self, simulation_configuration: UUID
+    ) -> pd.DataFrame:
         """Returns the spawn events by a given config id
         :param config_id: config id
         :return: spawn events dataframe
         """
-        raise NotImplementedError()
+        spawner_configurations = list(
+            SpawnerConfigurationXSimulationConfiguration.select().where(
+                SpawnerConfigurationXSimulationConfiguration.simulation_configuration
+                == simulation_configuration
+            )
+        )
+        demand_schedule_strategies_configurations = [
+            [
+                config.schedule_configuration_id
+                for config in SpawnerConfigurationXSchedule.select().where(
+                    SpawnerConfigurationXSchedule.spawner_configuration_id
+                    == spawner_config.spawner_configuration_id
+                )
+            ]
+            for spawner_config in spawner_configurations
+        ]
+        demand_schedule_strategies_configurations = [
+            item
+            for sublist in demand_schedule_strategies_configurations
+            for item in sublist
+        ]
+        demand_schedule_strategies_configurations = list(
+            ScheduleConfiguration.select().where(
+                (ScheduleConfiguration.id << demand_schedule_strategies_configurations)
+                & (ScheduleConfiguration.strategy_type == "DemandScheduleStrategy")
+            )
+        )
+        demand_schedule_strategies = [
+            (DemandScheduleStrategy.from_schedule_configuration(config), config.id)
+            for config in demand_schedule_strategies_configurations
+        ]
+        dataframes = []
+        for strategy, config_id in demand_schedule_strategies:
+            strategy.calculate_spawn_ticks()
```

Comment:
Why do you call it here? It was already called during the construction of the object. See line 81 in `src/schedule/demand_schedule_strategy.py`.

Reply:
Got it :)